### DISCUSSION TITLE
Technitium Performance Production Environment

### DISCUSSION THREAD

**Discussion by MichalisDBA** [TechnitiumSoftware/DnsServer#1125] [Q&A] [answered]
Hello.

We plan to use Technitium in our production environment but we have systems that make DNS requests in magnitude of millions per day. Let's 10mil DNS queries per day or more.

Can Technitium handle this load? Has anyone have a real experience with such high DNS traffic and Technitium?

**Comment by ShreyasZare** [comment]
Thanks for asking. The performance really depends on the server hardware, network, and type of queries. If the queries are mostly being served from cache then it would perform much better compared to queries which are mostly random causing frequent resolution.

For cached queries, the load testing done on a commodity desktop hardware with Gigabit Ethernet adapter can give around 100k req/sec maxing out the network adapter's packet per second bandwidth.

That said, I know of a deployment by a city ISP which has around 4 DNS instances setup and does around 100k-300k req/min on each of the instances. They have been using the DNS server since last 3 years by now.

You will need to update few of the options in settings for large deployments. Like increase or disable queries per minute option that is configured by default to do rate limiting. The Cache Maximum Entries option will need to be increased 10x or 20x so that the DNS server can hold more records in cache to improve performance.

**Reply by MichalisDBA** [reply]
Thank you for your answer. So you mean this city ISP uses 4 Technitium servers?

We are planing to use Technitium in Linux servers. I have some hesitation about running .NET in Linux and how much stable will it be. Any insides i have to look for by deploying it in Linux machines?

Thanks.

**Reply by ShreyasZare** [reply, marked as answer]
Yes, they use 4 DNS servers which were running on VMs in proxmox on two separate rack servers.

.NET on Linux has been very stable since many years. Its been like 8 years since the first release for .NET Core with Linux support. Its already being used in production by many companies. As an end user, you will not have to know anything about .NET for running Technitium DNS server. 

The only thing to remember is to not use any Linux distro which have officially run out of support since this causes issues when certain libraries stop being maintained for older Linux distros. For example, if you run the DNS server on Ubuntu server 18.04, it would work but the Query Logs (Sqlite) app wont works since the new sqlite native drivers do not support Ubuntu server 18.04 or older versions. There will also be issue with having support for `libmsquic` library (needed for QUIC protocol support) for older OS versions. So, its good to upgrade the OS every 2-3 years at max to avoid such issues.

**Reply by clintguy74** [reply]
Im currently running 5 servers for a large education facility, have been for the last 6 months, on linux and Sorry to burst anybodys bubble, but g**gle and clo-dflare eat your heart out.  I just hope technitium will eventually support enum, then i can use it for my entire network.

**Reply by ShreyasZare** [reply]
@clintguy74 Thanks for the comment. I am not sure what you mean by "support enum".

**Reply by clintguy74** [reply]
No problem, Ok, so basically I have been battling to get enum, to run, they say DNS servers can also be configured to support ENUM by handling the necessary NAPTR records. But I really have been battling, to get it to interact with my asterisk VOIP Server. is anybody you aware of that is using enum on technitium?
